x-pack/plugins/elastic_assistant/server/ai_assistant_service/create_resource_installation_helper.ts

```diff
@@ -109,13 +111,15 @@ export function createResourceInstallationHelper(
         );
       }
     },
-    getInitializedResources: async (namespace: string): Promise<InitializationPromise> => {
+    getInitializedResources: (namespace: string): Promise<InitializationPromise> => {
       const key = namespace;
-      return (
-        initializedResources.has(key)
-          ? initializedResources.get(key)
-          : errorResult(`Unrecognized spaceId ${key}`)
-      ) as InitializationPromise;
```

Comment:
- This function is not doing any async behaviour, it just retrieve a promise it set before.
- The type casting for the return value doesn't seem to be ideal